include/dr/mhp/algorithms/inclusive_exclusive_scan_impl.hpp

```diff
@@ -55,22 +69,37 @@ auto inclusive_scan_impl_(R &&r, O &&d_first, BinaryOp &&binary_op,
     if (dr::ranges::rank(global_in) == rank) {
       auto local_in = dr::ranges::__detail::local(global_in);
       auto local_out = dr::ranges::__detail::local(global_out);
+      if (is_exclusive && seg_index != 0) {
+        auto input_seg = input[seg_index - 1];
+        auto input_seg_size = rng::size(input_seg);
+        auto input_value = input_seg[input_seg_size - 1];
```

Comment:
Here there is communication with previous process. It can be avoid:

1. implement local_exclusive_scan in the way that on segment 0 it uses init value from user, on other it ignores it but copy the first element from input[0] to output[0] , then call exclusive scan starting from the second element and giving the first one as init value
2. specialize `back` variable computation in "Pass 2 put partial sums on root" in such a way that for exclusive scan it sums (operator-es) `local_out.back()` with `local_in.back()`

Reply:
As I said when we talked I fixed the algo and Done, thanks ;)